surveys/2024-annual-survey/questions.md

```diff
@@ -414,11 +382,28 @@ Type: select all that apply (optional)
 - Achieving structured concurrency with async code
 - Writing executor-agnostic async code
 - Borrow checker not allowing valid code
-- Other (please specify)
+- Slow runtime performance
+- Slow compile times
+- Achieving small binary size of compiled artifacts
+- Memory usage (example: how much RAM rustc uses when compiling)
+- Disk space usage (example: the size of target folder)
+- Bugs in the compiler (that is, ICEs a.k.a. internal compiler errors, miscompilations, etc.)
+- Opaque / unclear compiler error messages
+- Subpar IDE support
+- Debugging experience is not good enough (example: unclear / misleading diagnostics)
+- Package management (crates.io)
+- Rust language and standard library documentation
+
+How bad do I need this?:
```

Comment:
There isn't actually any label for this in SurveyHero, so this text ("How bad do I need this") won't appear anywhere. Which is good, because the wording would be weird ("Slow compile times" -> "How bad do I need this" is probably not a good lead-in :laughing: ). We could add some text just below the question text though.

So either we have to ask the main question in a way that the prepared answers make sense for it, or explain the answers below the question.

It seems to me that the answers in the vein of "it would do something for me" are too.. complicated, so that they make sense across all the options. What about something like this? It's hard to come up with good options for all four columns though..
![image](https://github.com/user-attachments/assets/96a14d67-7b88-4617-acea-f71a71f3ada4)

Reply:
How does SH recognize and separate in the matrix the block of answer options from the topics? "How bad do I need this" was just a funny block separator between the two but it can be removed (and def. should be reworded).

Choose an assessment only for the topics you think they are impacting your work. If you don't choose an answer, we will interpret as "I don't know" or that you are neutral on that topic.

About the column layout: now that I see it in the layout I'd say let's remove the "I don't know" option to win some horizontal space an try to reword the answer options as you suggest to keep them as short as possible.

Reply:
> How does SH recognize and separate in the matrix the block of answer options from the topics? "How bad do I need this" was just a funny block separator between the two but it can be removed (and def. should be reworded).

It doesn't, we fill in the questions into SH manually xD We do tend to include *some* separator in the Markdown file though, to recognize the split, as you said.

> About the column layout: now that I see it in the layout I'd say let's remove the "I don't know" option to win some horizontal space an try to reword the answer options as you suggest to keep them as short as possible.

Ok, we can just interpret a missing answer as "I don't know" (as the text already says), it will just need a bit of postprocessing.

Reply:
I have pushed a proposal to drastically shorten the colum width? How does it look?

Reply:
Having just three columns is :+1: But yeah, I'll need to think of the wording to unify it. "slow runtime performance" -> "very much needed" isn't ideal xD I'll try to rephrase all questions and the answers so that it makes more sense.

Reply:
> I'll try to rephrase all questions and the answers so that it makes more sense.

I'll also have a look at that